RoFILib/reconfig/rrt.cpp

```diff
@@ -0,0 +1,449 @@
+#include "Algorithms.h"
+#include "test/test_rrt.h"
+
+namespace Distance
+{
+    inline double reconnections(const Configuration& curr, const Configuration& goal) {
+        auto diff = curr.diff(goal);
+        return diff.reconnections().size();
+    }
+
+    inline double rotations(const Configuration& curr, const Configuration& goal) {
+        auto diff = curr.diff(goal);
+        return diff.rotations().size();
+    }
+
+    inline double diff(const Configuration& curr, const Configuration& goal) {
+        return reconnections(curr, goal) * 1000 + rotations(curr, goal);
+    }
+}
+
+inline int findFreeIndex(int index, const std::array<bool, 6>& A)
+{
+    int count = 6;
+    while (!A[index] && count > 0)
+    {
+        index = (index + 1) % 6;
+        --count;
+    }
+    if (count == 0)
+    {
+        return -1;
+    }
+    return index;
+}
+
+std::optional<Edge> generateEdge(ID id1, ID id2, std::unordered_map<ID, std::array<bool, 6>>& occupied)
```

Comment:
If I am not mistaken, `generateEdge` and `generateRotation` are RRT local functions -> add static to them.

Reply:
Taking back. They need to be visible for tests.